CRF reminder scheduling

Erin-Mounts opened the pull request:
seriously regretting having started from the JourneyPRO journey timeline code

Erin-Mounts committed:
CRF reminder scheduling

syoung-smallwisdom commented on CRFModuleValidation/MasterScheduledActivityManager.swift:
```diff
@@ -175,33 +175,47 @@ class MasterScheduledActivityManager: ScheduledActivityManager {
             
             // Remove previous reminders
-            let identifiers = self.schedules.map({ $0.taskGroup.identifier })
+            let identifiers = self.scheduleItemsToRemindOf().map({ $0.identifier })
```
You are only removing the schedules that have **not** been finished. What if the person does the task *before* the reminder notification fires?

Might be simpler (for the purposes of this study) to just use hardcoded identifiers and remove all.

Or to use a single identifier like "org.sagebase.CRFValidationApp" since then that reminder will be removed.

Alternatively, you could also just remove all reminders rather than trying to only remove those for which the notification is setup to fire.

syoung-smallwisdom reviewed (changes requested):
Need to fix how you are removing the identifiers.

Erin-Mounts committed:
respond to PR comment

Erin-Mounts committed:
on non-event days, don’t assume it’s past the end

…and thus end up showing the final clinic visit as “today’s activity”. Instead treat the previous day as the current section (but don’t call it “Today” in the UI).

Erin-Mounts committed:
Only remove Schedule-based reminders

also mark clinic1 or clinic2 task as finished rather than deleting (oops, though probably didn’t affect anything)

syoung-smallwisdom reviewed (approved):
looks good
